monitoring/monitoring.go

```diff
@@ -260,7 +260,15 @@ func Start(env e2e.Environment, opts ...Option) (_ *Service, err error) {
 	}))
 
 	// Listen on all addresses, since we need to connect to it from docker container.
-	list, err := net.Listen("tcp", "0.0.0.0:0")
+	networkType := "tcp"
+	// For unknown reasons, when using WSL 2, if the network type is "tcp" it will
+	// end up only binding to the IPv6 in the WSL host, which later cannot be acessed
+	// via IPv4 to confirm Prometheus can scrape the local endpoint.
+	// Explicitly asking for an IPv4 listener works.
+	if env.WSL2() {
+		networkType = "tcp4"
```

Comment:
Is tcp4 the more specific option (specifies you want an ipv4 connection)? If not we can just make that the default and get rid of the branching conditionals.

Reply:
We can do this. Only downside would be that Docker one day might drop support for ipv4, but I bet that's likely going to happen only after our lifetimes. Will change!

Reply:
Oh, this might also not work on servers with only ipv6 stack. 😬

Reply:
What do you think, @moadz?

Reply:
Can we use IPv6 then for network port later on? If the latter step requires IPv4, we might as well put tcp4 here.

Reply:
This is only affecting the listener that is used for a quick test of the host-container networking. It won't affect anything after this point.